tools/scalajsenv.js

```diff
@@ -180,11 +180,11 @@ ScalaJS.isScalaJSObject = function(obj) {
 //!if asInstanceOfs != Unchecked
 ScalaJS.throwClassCastException = function(instance, classFullName) {
 //!if asInstanceOfs == Compliant
-  throw new ScalaJS.c.jl_ClassCastException().init___T(
+  throw new ScalaJS.c.jl_ClassCastException(
     instance + " is not an instance of " + classFullName);
```

Comment:
This is dangerous. `ClassCastException` has two constructors (one without any parameter). Apparently, this happens to work because the no-arg constructor is not called by anything in the test suite, and therefore it is thrown off by the Analyzer. But a code base containing a `new ClassCastException()` would blow up.

Fixing this is hard, as we cannot know from `scalajsenv.js` whether the second constructor was kept or discarded. Worse, `ClassCastException` is not even `final` (at the Java level), which means it is, in theory, possible to have a codebase with a subclass of `ClassCastException`.

I do not know how to address this yet. I will think about it.